lib/video_info/providers/youtube_scraper.rb

```diff
@@ -76,13 +77,17 @@ def _set_data_from_api(api_url = _api_url)
           uri.scheme = 'http'
         end
 
-        # handle fullscreen video URLs
-        if url.include?('.com/v/')
-          video_id = url.split('/v/')[1].split('?')[0]
-          new_url = 'https://www.youtube.com/watch?v=' + video_id
-          Oga.parse_html(open(new_url).read)
-        else
-          Oga.parse_html(open(uri.to_s, allow_redirections: :safe).read)
+        begin
+          # handle fullscreen video URLs
+          if url.include?('.com/v/')
+            video_id = url.split('/v/')[1].split('?')[0]
+            new_url = 'https://www.youtube.com/watch?v=' + video_id
+            Oga.parse_html(open(new_url).read)
+          else
+            Oga.parse_html(open(uri.to_s, allow_redirections: :safe).read)
+          end
+        rescue OpenURI::HTTPError, *NetHttpTimeoutErrors.all
+          raise VideoInfo::HttpError.new "unexpected network error while fetching information about the video"
```

Comment:
Line is too long. [110/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.